frame/staking/src/pallet/mod.rs

```diff
@@ -813,7 +818,12 @@ pub mod pallet {
 				total: value,
 				active: value,
 				unlocking: Default::default(),
-				claimed_rewards: (last_reward_era..current_era).collect(),
+				claimed_rewards: (last_reward_era..current_era)
```

Comment:
would be nice to make sure this is checked with some reasonable values in tests that test the `bond` transaction. Good reason to play around with a few of the tests as well.

Reply:
Some tests were already testing these values. I refactored them a bit to read/calculate first and last reward era in code. I also changed HistoryDepth value in staking mock runtime to 80 to keep it a bit different from actual runtime and make sure in test we are not still reading the hardcoded value `84` like earlier.

Reply:
Actually none of them is testing the claimed rewards after a bond transaction. I will add that.